File: 1816-truncate-sentence/1816-truncate-sentence.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var truncateSentence = function(s, k)  {
    
    let spaces = 0;
    for(let i = 0 ; i < s.length; i ++){
        if(s[i] === ' ' && ++spaces === k){
            return s.slice(0,i)
        }
    }
    
    return s;
    
};

File: 1816-truncate-sentence/1816-truncate-sentence.java
class Solution {
    public String truncateSentence(String s, int k) {
        
        
        int spaces = 0;
        int length = s.length();
        int index = 0;
        
       while(index < length){
               if(s.charAt(index) == ' '){
                   spaces++;
               }
           if(spaces == k){
               return s.substring(0,index);
           }
           index++;
           
       }
        
    return s;
        
    }
}